soroban-env-host/benches/common/measure.rs

```diff
@@ -292,34 +289,44 @@ where
     let mut alloc_group_token =
         AllocationGroupToken::register().expect("failed to register allocation group");
     let mut ret = Vec::new();
-    eprintln!("\nMeasuring costs for CostType::{:?}\n", HCM::COST_TYPE);
+    eprintln!(
+        "\nMeasuring costs for CostType::{:?}\n",
+        <HCM::Runner as CostRunner>::COST_TYPE
+    );
     loop {
+        // prepare the measurement
         let host = Host::default();
         host.with_budget(|budget| budget.reset_unlimited());
-        let mut m = match next_hcm(&host) {
+        // memory tracking needs to start here (instead of right before `run`) because the sample
+        // data container allocation needs to be counted. Otherwise, the allocation/deallocation
+        // count will be asymmetric, which will lead to mem tracker count underflow.
```

Comment:
@graydon Please see this comment, and my comment with this PR.

Reply:
@graydon and I discussed and decided we only count allocations not deallocations since we care about the worst-case estimate, and this mimics the arena model which is the worst case. I updated the `deallocated` and added a comment. These two lines are moved back to where it was before (right before running the host).